### PR TITLE
Add duplicate_with for RSpec Example

### PR DESCRIPTION
**Pull request by bootstraponline**
This is the last of the new APIs as discussed in [#2094](https://github.com/rspec/rspec-core/pull/2094#issuecomment-149746720)

### CODE REVIEW

**Comment by JonRowe** [comment]
I love how simple this is!

**Comment by bootstraponline** [comment]
Contributing to rspec-core has been a great experience. Thanks for the timely feedback and code reviews.

File: lib/rspec/core/example.rb [hunk 1]
```diff
@@ -124,6 +124,22 @@ def self.parse_id(id)
...
+      # @param metadata_overrides [Hash] the hash to override the example metadata
+      # @return [Example] a duplicate of the example with modified metadata
+      def duplicate_with(metadata_overrides={})
+        new_metadata = metadata.clone.merge(metadata_overrides)
+
+        RSpec::Core::Metadata::RESERVED_KEYS.each do |reserved_key|
+          new_metadata.delete reserved_key
+        end
+
+        Example.new(example_group.clone, description.clone,
```
Comment (myronmarston) [inline lib/rspec/core/example.rb]: The example group should not be cloned.  The example should belong to the _same_ example group class, not a copy of it. The clone is likely to cause some bugs.

Comment (bootstraponline) [inline lib/rspec/core/example.rb]: Nice catch. I'll submit a fix & update the spec to catch this bug.

**Pull request merged**